### PR TITLE
feat(WalletConnect): Fine-tune connect dApp modal

### LINKED ISSUE #14690: [UI - dApps Connection]  Create `Connection request` view that will be part of `Connect a dApp` dialog
[Design](https://www.figma.com/file/HrmZp1y4S77QJezRFRl6ku/Descoped-dApp-Interactions?type=design&node-id=481%3A160572&mode=design&t=iMRTBgKquR8Eq9bf-1)

Screenshot 2024-05-10 at 10 19 36

- [x] Base view components:
  - Implemented using `layer.effect: OpacityMask { maskSource: Rectangle`
  - Image + badge image: Maybe it needs a new SQ component for that
 **Image fetched**:
Screenshot 2024-05-10 at 13 47 53
  - Low priority
    - [ ] The wallet-connect sub-icon

- [x] **Deafult image if data not fetched**:
Screenshot 2024-05-10 at 13 48 44

- [ ] Subtitle + link icon --> all these are must be clickable, not only the icon
Screenshot 2024-05-10 at 13 50 07

- [ ] Section with `AccountsSelector`component. Use the one build in [here](https://github.com/status-im/status-desktop/issues/14691)
   - Using `SendModal`'s header account chooser for the intial implementation

- [ ] If `dApp` is already connected select that account

- [ ] Section with `NetworkFilter` component. Use the one build in [here](https://github.com/status-im/status-desktop/issues/14692)
  - Using the normal network selector for the initial implementation with disabled option for choosing networks

- [x] **Connect transition**:
  - New tag with the status of the connection (Success or failed) + [animation](https://www.figma.com/file/HrmZp1y4S77QJezRFRl6ku/Descoped-dApp-Interactions?type=design&node-id=481%3A165751&mode=design&t=iMRTBgKquR8Eq9bf-1) for 2 seconds on the color of the tag 
Screenshot 2024-05-10 at 13 55 50
Screenshot 2024-05-10 at 13 55 56

  - Some texts changed to reflect the new status
  - Footer buttons changed according to new status 
     - Decline / Close if connected
     - Close if error

- [x] **Decline transition**: It raises an [ephemeral notification](https://www.figma.com/file/HrmZp1y4S77QJezRFRl6ku/Descoped-dApp-Interactions?type=design&node-id=0%3A1&mode=design&t=iMRTBgKquR8Eq9bf-1)
Screenshot 2024-05-10 at 13 53 31

- [ ] What `description` and `icons` to use for the app? See `metadata` parameter in `Web3Wallet.init` call
    - Started discussion here: https://discord.com/channels/1210237582470807632/1217175655729467492/1242744084964179990

### PR DESCRIPTION
**Pull request by alexjba**
### What does the PR do

Closes #14690

Changes:
1. Align dialog with Figma design
2. Add new components for round image with badge and connection status tag
3. Add tests
4. Dapps service will now receive wallet `RootStore` as input and reuse existing models with account balance and other necessary info for account selection and chain selection
5. Minor updates in stores
6. Minor updates in WC toast messages to display app domain instead of app url
7. Minor updates to accommodate larger text

### Affected areas

Wallet connect

### Screenshot of functionality (including design for comparison)

https://github.com/status-im/status-desktop/assets/47811206/c880d88c-526e-41bf-8a32-05fac0ace750

- [x] I've checked the design and this PR matches it

### CODE REVIEW

**Comment by alexjba** [comment]
UI tests constantly crashing..Maybe the inline components again 👀

File: ui/app/AppLayouts/Wallet/services/dapps/DAppsRequestHandler.qml [hunk 1]
```diff
@@ -189,8 +190,9 @@ QObject {
                 }
                 address = event.params.request.params[0].from
             }
-            return ModelUtils.getFirstModelEntryIf(walletStore.ownAccounts, (account) => {
-                return account.address.toLowerCase() === address.toLowerCase()
+            return ModelUtils.getFirstModelEntryIf(root.accountsModel, (account) => {
+                print ("!!!!!!", account.address.toLowerCase(), address.toLowerCase())
```
Comment (stefandunca) [inline ui/app/AppLayouts/Wallet/services/dapps/DAppsRequestHandler.qml:194]: Leftovers?

File: ui/imports/shared/popups/walletconnect/ConnectDAppModal.qml [hunk 2]
```diff
@@ -21,73 +21,84 @@ import shared.popups.send.controls 1.0
...
 import utils 1.0
+import shared.popups.walletconnect.private 1.0
 
 StatusDialog {
     id: root
 
-    width: 480
-    implicitHeight: d.connectionStatus === root.notConnectedStatus ? 633 : 681
-
+    /*
```
Comment (stefandunca) [inline ui/imports/shared/popups/walletconnect/ConnectDAppModal.qml:29]: 👌

File: ui/imports/shared/popups/walletconnect/ConnectDAppModal.qml [hunk 3]
```diff
@@ -400,7 +218,17 @@ StatusDialog {
...
         readonly property var dappChains: ListModel {}
 
-        property int connectionStatus: notConnectedStatus
-        property bool afterTwoSecondsFromStatus: false
+        property int connectionStatus: root.notConnectedStatus
+        readonly property bool connectionSuccessful: d.connectionStatus === root.connectionSuccessfulStatus
+        readonly property bool connectionFailed: d.connectionStatus === root.connectionFailedStatus
+        readonly property bool connectionAttempted: d.connectionStatus !== root.notConnectedStatus
+
+        function syncDAppChainsModel() {
```
Comment (stefandunca) [inline ui/imports/shared/popups/walletconnect/ConnectDAppModal.qml:226]: I'm wondering why are we using a manual sync and not a proxy model.

File: ui/imports/shared/popups/walletconnect/private/ContextCard.qml [hunk 4]
```diff
@@ -0,0 +1,87 @@
+import QtQuick 2.15
```
Comment (stefandunca) [inline ui/imports/shared/popups/walletconnect/private/ContextCard.qml:1]: Note that @Seitseman already extracted these in his PR.

**Review by stefandunca — APPROVED** [review]
I love seeing the UX polished

File: ui/imports/shared/popups/walletconnect/private/ContextCard.qml (hunk 4 above)
Comment (alexjba) [reply ui/imports/shared/popups/walletconnect/private/ContextCard.qml:1]: ohh, right! In this case I'll rebase on top of his PR

File: ui/imports/shared/popups/walletconnect/ConnectionStatusTag.qml [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+import shared.controls 1.0
+import utils 1.0
+
+InformationTag {
+    id: root
+
+    property bool success: false
+
+    tagPrimaryLabel.text: qsTr("Connected. You can now go back to the dApp.") //{
```
Comment (caybro) [inline ui/imports/shared/popups/walletconnect/ConnectionStatusTag.qml:13]: some cruft at the end of the line

File: ui/imports/shared/popups/walletconnect/RoundImageWithBadge.qml
Comment (caybro) [inline ui/imports/shared/popups/walletconnect/RoundImageWithBadge.qml:1]: You could achieve the same thing with a `StatusSmartIdenticon`, albeit a bit more bloated

File: ui/imports/shared/popups/walletconnect/ConnectDAppModal.qml (hunk 3 above)
Comment (alexjba) [reply ui/imports/shared/popups/walletconnect/ConnectDAppModal.qml]: Good point, thanks! I've mindlessly kept the dappChains in the API. This component doesn't need to know about it. Removed it from the API and replaced with a SFPM filtering the flatNetworks to show only the supported chains.

**Pull request merged**